### PR TITLE
Feature/deps: fix simulation error

### PR DESCRIPTION
**Pull request by kaifei-bianjie**

### CODE REVIEW

File: app/sim_test.go [hunk 1]
```diff
@@ -29,58 +32,70 @@ var (
...
 
 func init() {
 	flag.Int64Var(&seed, "SimulationSeed", 42, "Simulation random seed")
-	flag.IntVar(&numBlocks, "SimulationNumBlocks", 500, "Number of blocks")
+	flag.IntVar(&numBlocks, "SimulationNumBlocks", 50, "Number of blocks")
 	flag.IntVar(&blockSize, "SimulationBlockSize", 200, "Operations per block")
-	flag.BoolVar(&enabled, "SimulationEnabled", false, "Enable the simulation")
+	flag.BoolVar(&enabled, "SimulationEnabled", true, "Enable the simulation")
 	flag.BoolVar(&verbose, "SimulationVerbose", false, "Verbose log output")
+	flag.BoolVar(&commit, "SimulationCommit", false, "Have the simulation commit")
```
Comment (chengwenxi) [inline app/sim_test.go]: SimulationNumBlocks & SimulationEnabled should not be modified, if SimulationEnabled is true, unit test may deal with sim test

File: app/sim_test.go [hunk 2]
```diff
@@ -92,31 +107,58 @@ func appStateFn(r *rand.Rand, keys []crypto.PrivKey, accs []sdk.AccAddress) json
...
-			banksim.NonnegativeBalanceInvariant(app.AccountKeeper)(t, baseapp, log)
-			govsim.AllInvariants()(t, baseapp, log)
-			stakesim.AllInvariants(app.coinKeeper, app.stakeKeeper, app.AccountKeeper)(t, baseapp, log)
-			slashingsim.AllInvariants()(t, baseapp, log)
-		},
+	return []simulation.Invariant{}
+}
+
+// Profile with:
+// /usr/local/go/bin/go test -benchmem -run=^$ github.com/cosmos/cosmos-sdk/cmd/gaia/app -bench ^BenchmarkFullGaiaSimulation$ -SimulationCommit=true -cpuprofile cpu.out
```
Comment (chengwenxi) [inline app/sim_test.go]: Should replace gaia to iris

File: app/sim_test.go [hunk 3]
```diff
@@ -125,7 +167,7 @@ func TestFullIrisSimulation(t *testing.T) {
 		t.Skip("Skipping Iris simulation")
 	}
 
-	// Setup Iris application
+	// Setup Gaia application
```
Comment (chengwenxi) [inline app/sim_test.go]: Iris

File: app/sim_test.go [hunk 4]
```diff
@@ -146,17 +188,20 @@ func TestFullIrisSimulation(t *testing.T) {
...
-
+	if commit {
+		fmt.Println("Database Size", db.Stats()["database.size"])
+	}
+	require.Nil(t, err)
 }
 
 // TODO: Make another test for the fuzzer itself, which just has noOp txs
-// and doesn't depend on iris
+// and doesn't depend on gaia
```
Comment (chengwenxi) [inline app/sim_test.go]: iris

File: app/sim_test.go (hunk 2 above)
Comment (chengwenxi) [reply app/sim_test.go]: Add benchmem command in Makefile?

File: app/sim_test.go [hunk 5]
```diff
@@ -92,31 +107,58 @@ func appStateFn(r *rand.Rand, keys []crypto.PrivKey, accs []sdk.AccAddress) json
...
-			stakesim.AllInvariants(app.coinKeeper, app.stakeKeeper, app.AccountKeeper)(t, baseapp, log)
-			slashingsim.AllInvariants()(t, baseapp, log)
-		},
+	return []simulation.Invariant{}
+}
+
+// Profile with:
+// /usr/local/go/bin/go test -benchmem -run=^$ github.com/cosmos/cosmos-sdk/cmd/gaia/app -bench ^BenchmarkFullGaiaSimulation$ -SimulationCommit=true -cpuprofile cpu.out
+func BenchmarkFullIrisSimulation(b *testing.B) {
+	// Setup Gaia application
```
Comment (chengwenxi) [inline app/sim_test.go]: gaia -> iris

File: app/sim_test.go (hunk 1 above)
Comment (kaifei-bianjie) [reply app/sim_test.go]: ok, i will set SimulationEnabled = false

File: app/sim_test.go (hunk 2 above)
Comment (kaifei-bianjie) [reply app/sim_test.go]: > Add benchmem command in Makefile?

not add benchmem command now

**Pull request merged**